src/cljc/athens/common_events/resolver.cljc

```diff
@@ -648,6 +648,54 @@
     tx-data))
 
 
+;; resetting ds-conn re-computes all subs and is the cause
+;;    for huge delays when undo/redo is pressed
+;; here is another alternative strategy for undo/redo
+;; Core ideas are inspired from here https://tonsky.me/blog/datascript-internals/
+;;    1. db-before + tx-data = db-after
+;;    2. DataScript DB contains only currently relevant datoms.
+;; 1 is the math that is happening here when you undo/redo but in a much more performant way
+;; 2 asserts that even if you are reapplying same txn over and over only relevant ones will be present
+;;    - and overall size of transit file does not increase
+;; Note: only relevant txns(ones that user deliberately made, not undo/redo ones) go into history
+;; Note: Once session is lost(App is closed) edit history is also lost
+;; Also cmd + z -> cmd + z -> edit something --- future(cmd + shift + z) doesn't work
```

Comment:
For that to work we'd need to have something like https://www.emacswiki.org/emacs/UndoTree, but even then last command wouldn't work because "edit something" introduces a fork in undo tree.

Reply:
Just to be clear, I don't mean we should be doing it here, it's just for reference.

Reply:
I'll check that. This comment is from the previous code. I copied it here to preserve the info.